app/src/main/kotlin/com/simplemobiletools/clock/helpers/DBHelper.kt

```diff
@@ -47,7 +48,13 @@ class DBHelper private constructor(val context: Context) : SQLiteOpenHelper(cont
         insertInitialAlarms(db)
     }
 
-    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}
+    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
+        var version = oldVersion
```

Comment:
why do we need this variable? We already have old and new version as parameters

Reply:
We don't really need it now. I can remove it, but I meant it more as a useful thing to have when future migrations are added, because then all migrations can be added one by one:

```
var version = oldVersion // needed because we need to modify it as we migrate
if (version == 1 && newVersion > version) { // Current version is 1 and we are migrating to some higher version - do 1_2 migration
  // migrate
  version++
}
if (version == 2 && newVersion > version) { // If we are at version 2 and upgrading - this will be called when migrating from version 1 too, since we added that version++ there
  // migrate
  version++
}
// Keep adding new migrations here
```

Reply:
migrations are already done one by one, this variable is always useless

Reply:
How do you mean?
This method is called only once and only the current database version (the one available on the device at the time of calling) is passed and new version (which is the latest one, that we defined in the constructor).

If the device currently has version 1 and user has missed a couple of updates and downloads a version of the app that used version 4, this method will only be called once with `oldVersion = 1` and `newVersion = 4`. In that case we should invoke all migrations (`1 -> 2`, `2 -> 3`, `3 -> 4`).

Either way, I will remove this variable now, since it is not required at the moment, but we need to be careful if we upgrade this version once again, to ensure that we cover case like the one above.

Reply:
well you can already check old and new version, those are enough